runtime-light/coroutine/shared-task.h

```diff
@@ -129,28 +129,22 @@ struct promise_base_t {
     const auto *waiter_ptr{std::addressof(waiter)};
     if (m_waiters == waiter_ptr) { // waiter is the head of the list
       m_waiters = waiter_ptr->m_next;
-      return;
+    } else if (waiter_ptr->m_next == nullptr) { // waiter is the last in the list
+      waiter_ptr->m_prev->m_next = nullptr;
+    } else { // waiter is somewhere in the middle of the list
+      waiter_ptr->m_next->m_prev = waiter_ptr->m_prev;
+      waiter_ptr->m_prev->m_next = waiter_ptr->m_next;
```

Comment:
Can't `waiter_ptr->m_prev` be nullptr here?

Reply:
No, it can't since the waiter here is somewhere in the middle of the waiters list, so it has both prev and next